File: frontend/app/src/main/java/com/example/test/RatingAdapter.java
package com.example.test;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class RatingAdapter extends RecyclerView.Adapter<RatingAdapter.ViewHolder> {

    private final LayoutInflater inflater;
    private final List<RatingItem> ratingItems;

    RatingAdapter(Context context, List<RatingItem> ratingItems) {
        this.ratingItems = ratingItems;
        this.inflater = LayoutInflater.from(context);
    }

    @Override
    public RatingAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = inflater.inflate(R.layout.row_rating, parent, false);
        return new ViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(RatingAdapter.ViewHolder holder, int position) {
        RatingItem ratingItem = ratingItems.get(position);
        holder.positionView.setText(Integer.toString(position));
        holder.nameView.setText(ratingItem.getFirstName() + " " + ratingItem.getSecondName());
        holder.scoreView.setText(Integer.toString(ratingItem.getScore()));
    }

    @Override
    public int getItemCount() {
        return ratingItems.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView positionView, nameView, scoreView;

        ViewHolder(View view) {
            super(view);
            positionView = view.findViewById(R.id.txtPosition);
            nameView = view.findViewById(R.id.txtName);
            scoreView = view.findViewById(R.id.txtScore);
        }
    }
}

File: frontend/app/src/main/java/com/daniiltropolsapplication/app/modules/rating/ui/RatingAdapter.kt
package com.daniiltropolsapplication.app.modules.rating.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.daniiltropolsapplication.app.R
import com.daniiltropolsapplication.app.databinding.RowRatingBinding
import com.daniiltropolsapplication.app.modules.rating.`data`.model.RatingRowModel
import kotlin.Int
import kotlin.collections.List

class RatingAdapter(
  var list: List<RatingRowModel>
) : RecyclerView.Adapter<RatingAdapter.RowRatingVH>() {
  private var clickListener: OnItemClickListener? = null

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RowRatingVH {
    val view=LayoutInflater.from(parent.context).inflate(R.layout.row_rating,parent,false)
    return RowRatingVH(view)
  }

  override fun onBindViewHolder(holder: RowRatingVH, position: Int) {
    val ratingRowModel = RatingRowModel()
    // TODO uncomment following line after integration with data source
    // val ratingRowModel = list[position]
    holder.binding.ratingRowModel = ratingRowModel
  }

  override fun getItemCount(): Int = 4
  // TODO uncomment following line after integration with data source
  // return list.size

  public fun updateData(newData: List<RatingRowModel>) {
    list = newData
    notifyDataSetChanged()
  }

  fun setOnItemClickListener(clickListener: OnItemClickListener) {
    this.clickListener = clickListener
  }

  interface OnItemClickListener {
    fun onItemClick(
      view: View,
      position: Int,
      item: RatingRowModel
    ) {
    }
  }

  inner class RowRatingVH(
    view: View
  ) : RecyclerView.ViewHolder(view) {
    val binding: RowRatingBinding = RowRatingBinding.bind(itemView)
  }
}
